src/app/queries/hooks.ts

```diff
@@ -0,0 +1,126 @@
+import { MutationResultPair, QueryResult, useQueryCache } from 'react-query';
+import * as yup from 'yup';
+
+import { checkIfResourceExists, ForkliftResource, ForkliftResourceKind } from '@app/client/helpers';
+import { IKubeList, IKubeResponse, IKubeStatus, KubeClientError } from '@app/client/types';
+import { dnsLabelNameSchema, META } from '@app/common/constants';
+import { usePollingContext } from '@app/common/context';
+import {
+  mockKubeList,
+  sortKubeResultsByName,
+  useMockableMutation,
+  useMockableQuery,
+} from './helpers';
+import { MOCK_HOOKS } from './mocks/hooks.mock';
+import { IHook, IMetaObjectMeta } from './types';
+import { useAuthorizedK8sClient } from './fetchHelpers';
+import { HookFormState } from '@app/Hooks/components/AddEditHookModal';
+import { generateHook } from '@app/Hooks/components/helpers';
+
+const hookResource = new ForkliftResource(ForkliftResourceKind.Hook, META.namespace);
+
+export const useHooksQuery = (): QueryResult<IKubeList<IHook>> => {
+  const client = useAuthorizedK8sClient();
+  const result = useMockableQuery<IKubeList<IHook>>(
+    {
+      queryKey: 'hooks',
+      queryFn: async () => (await client.list<IKubeList<IHook>>(hookResource)).data,
+      config: {
+        refetchInterval: usePollingContext().refetchInterval,
+      },
+    },
+    mockKubeList(MOCK_HOOKS, 'Hook')
+  );
+  return sortKubeResultsByName(result);
+};
+
+export const useCreateHookMutation = (
```

Comment:
This is all great, thanks for knocking it out! It is making me think we could use some more abstraction on these basic CRUD hooks, but I'm getting ahead of myself again. ☕ ☕

Reply:
Well, that's it, the more I help you, the further you're go. Let's continue... ;)